src/enterprise/extension.ts

```diff
@@ -133,6 +137,7 @@ function initSelectionHandling(): vscode.Disposable {
 }
 
 export async function deactivate(): Promise<unknown> {
+  cancelWorkspaceUpdater();
   return requestDeactivate();
```

Comment:
Maybe this should go inside `requestDeactivate`?
And maybe there's also some common code in activate between enterprise and saas?

Reply:
> Maybe this should go inside `requestDeactivate`?

This API only deactivates the binary process. Other intervals are canceled outside this function.

>  And maybe there's also some common code in activate between enterprise and saas?

Unless @dimacodota knows something else, I didn't find such common code (maybe worth doing it, but not in this pr)

Reply:
id suggest to convert the startWorkspaceUpdater  to a WorkspaceUpdater that implements `Disposable `
and then add it to `context.subscriptions` this will handle all the cleanup 

see https://github.com/codota/tabnine-vscode/blob/77451b9438459ed8db2e56e13a40e3d9dde432e9/src/enterprise/statusBar/StatusBar.ts#L16 fotr exmple